packages/flutter/lib/src/gestures/velocity_tracker.dart

```diff
@@ -251,3 +251,86 @@ class VelocityTracker {
     return Velocity(pixelsPerSecond: estimate.pixelsPerSecond);
   }
 }
+
+/// A [VelocityTracker] subclass that matches iOS scroll view's fling velocity
+/// calculation.
+///
+/// To obtain a velocity, call [getVelocity] or [getVelocityEstimate]. The
+/// esimated velocity is typically used as the initial flinging velocity of a
+/// `Scrollable`, when its drag gesture ends.
+class FreeScrollStartVelocityTracker extends VelocityTracker {
+  /// The velocity estimation uses at most 4 samples. The extra samples are
```

Comment:
Not sure I understood this comment

Reply:
also seems like you're always using exactly 3 samples no?

Reply:
"4 `_PointAtTime` samples" is what I meant

Reply:
`Horizontal/VerticalDragGestureRecognizer.isFlingGesture` has a threshold for `VelocityEstimate.offset` for it to be a fling:
https://api.flutter.dev/flutter/gestures/VerticalDragGestureRecognizer/isFlingGesture.html

Initially I used 4 samples and that broke some tests so I had to record more points. Will point to `VerticalDragGestureRecognizer.isFlingGesture` in the comments.